libraries/core_libs/consensus/src/transaction/transaction_manager.cpp

```diff
@@ -349,4 +353,18 @@ bool TransactionManager::checkBlockTransactions(DagBlock const &blk) {
   return true;
 }
 
+void TransactionManager::blockFinalized(uint64_t block_number) {
+  std::unique_lock transactions_lock(transactions_mutex_);
+  auto &insufficient_balance_transactions = transactions_pool_.getInsufficientBalanceTransactions();
```

Comment:
here you are processing  `insufficient_balance_transactions_` after block is finalized. Where are you processing `low_nonce_transactions_` ? I dont see anything except saving txs there and then removing them after they expire

Reply:
low nonce transactions can never become valid because nonce is only increased, never reduced. We only need them to support any dag block that might have included them at the point that the node that created dag block thought they were valid

Reply:
ah ok